src/console.cpp

```diff
@@ -195,80 +204,36 @@ bool GetArgumentInteger(uint32 *value, const char *arg)
 }
 
 /**
- * Add an item to an alphabetically sorted list.
- * @param base first item of the list
- * @param item_new the item to add
+ * Creates a copy of a string with underscores removed from it
+ * @param name String to remove the underscores from.
+ * @return A copy of \a name, without underscores.
  */
-template<class T>
-void IConsoleAddSorted(T **base, T *item_new)
+static std::string RemoveUnderscores(const std::string &name)
 {
-	if (*base == nullptr) {
-		*base = item_new;
-		return;
-	}
-
-	T *item_before = nullptr;
-	T *item = *base;
-	/* The list is alphabetically sorted, insert the new item at the correct location */
-	while (item != nullptr) {
-		if (strcmp(item->name, item_new->name) > 0) break; // insert here
-
-		item_before = item;
-		item = item->next;
-	}
-
-	if (item_before == nullptr) {
-		*base = item_new;
-	} else {
-		item_before->next = item_new;
-	}
-
-	item_new->next = item;
-}
-
-/**
- * Remove underscores from a string; the string will be modified!
- * @param[in,out] name String to remove the underscores from.
- * @return \a name, with its contents modified.
- */
-char *RemoveUnderscores(char *name)
-{
-	char *q = name;
-	for (const char *p = name; *p != '\0'; p++) {
-		if (*p != '_') *q++ = *p;
-	}
-	*q = '\0';
-	return name;
+	std::string copy(name);
```

Comment:
Since a copy of the object is mandatory to be able to modify it, why using a reference to `name` in the function's signature?
Wouldn't `std::string RemoveUnderscores(std::string name)` be better?

Reply:
Done.